### ISSUE TITLE
EntityDecoder derived by argonaut.DecodeJson eats umlauts

### ISSUE THREAD

**Issue by mahe-ymc** [http4s/http4s#514]
Something like this reproduces the issue:
```
val x = Response().withBody(jsonString).map(x => decoder.decode(x.withContentType(Some(`application/json`)), true)).run.run.run
val y = Parse.decode[MyObject](jsonString)
```
Root of the problem seems to be the used jawnDecoder. After `msg.body.parseJson` the Umlauts are eaten.

Could be related to https://github.com/non/jawn/issues/45

**Comment by rossabaker** [comment]
I agree with the jawn analysis.  It doesn't happen without the escaped string.

**Comment by 2chilled** [comment]
What's the plan to handle this issue? Waiting until it's resolved upstream?

**Comment by rossabaker** [comment]
I can't think of a good workaround, and there is already a proposed solution upstream.  I'll bump the ticket.  If we can't make any progress there we can think about a fallback.  (Vendor branch? Shade the dependency? Ugh.)

**Comment by 2chilled** [comment]
I'm afraid that bumping won't fix this very soon as the solution is hanging there for two months now. A bad workaround would be to just build upon the EntityDecoder[String] to derive EntityDecoder[T] of DecodeJson[T]. Unfortunately this ticket is quite critical for us so I'm going to replace all jsonOf usages with an implementation of that idea in our project.

**Comment by rossabaker** [comment]
Another idea is to internally publish a bugfix of jawn and implement on top of that.  That's worth it if you are relying on the incremental parsing, and your idea is better if you're not.

**Comment by 2chilled** [comment]
Hmm, in my case publishing internally could be even more time efficient. Anyway, thanks for the support and even more for this great library!

**Comment by rossabaker** [comment]
A fixed jawn is being released tonight.  I'll follow that with a new jawnstreamz, and follow that with a patched http4s.

**Comment by rossabaker** [comment]
Anybody suffering from this should be able to fix it now with an explicit dependency on jawn-0.8.4.  A release to fix transient dependencies is coming.

**Comment by rossabaker** [comment]
Fixed in v0.12.3.
